core/common/src/main/java/alluxio/collections/ConcurrentHashSet.java

```diff
@@ -104,6 +104,9 @@ public boolean equals(Object o) {
       return true;
     }
     // allow equality with any Set.
+    if (!(o instanceof ConcurrentHashSet)) {
```

Comment:
@aaudiber I think this is necessary. Because `keySet()` return a `KeySetView<K,V>` for the concurrent HashMap we are using, which is not really associated with the class type of `ConcurrentHashSet`. In other words, if you implement a class `MyNewConcurrentHashSet` which return the same type of `KeySetView<K,V>` in this class, it will pass the equality test.